sign/dilithium/mode/mode.go

```diff
@@ -0,0 +1,66 @@
+// mode defines the interface for the various modes of the Dilithium
+// signature scheme modes.
+package mode
+
+import (
+	"io"
+)
+
+// PublicKey is a Dilithium public key.
+type PublicKey interface {
+	// Packs public key
+	Bytes() []byte
+}
+
+// PrivateKey is a Dilithium public key.
+type PrivateKey interface {
+	// Packs private key
+	Bytes() []byte
+}
+
+// Mode is a certain configuration of the Dilithium signature scheme.
+type Mode interface {
+	// GenerateKey generates a public/private key pair using entropy from rand.
+	// If rand is nil, crypto/rand.Reader will be used.
+	GenerateKey(rand io.Reader) (PublicKey, PrivateKey, error)
+
+	// NewKeyFromSeed derives a public/private key pair using the given seed.
+	// Panics if len(seed) != SeedSize()
+	NewKeyFromSeed(seed []byte) (PublicKey, PrivateKey)
```

Comment:
I was curious if there is needed a method that returns the seed.

Reply:
Sorry, I don't understand.  A method on the `PrivateKey`?  Or do you mean why this function exists at all?

Reply:
Suppose the following scenario:
Alice generates a key pair using GenerateKey(nil) -> sk,pk.
Then, Alice knows that she can get the same keys using NewKeyFromSeed if the seed is provided.
So, how does she can retrieve the seed previously generated in order to use NewKeyFromSeed?

Reply:
Ok, this is a bit tricky.

Preferably we would add a `func Seed() [SeedSize]byte` function on `PrivateKey` to return the seed that was used to generate it. But that requires us to store the seed inside the private key.  However, the binary format of the Dilithium private key does not store the seed, so we can't store it. We could store it in the struct anyway, but then we have a `Seed()` function which only works in certain situations (namely when it has just been created instead of unmarshaled).

Reply:
Is this still an issue?

Reply:
Maybe the solution is to hide `NewKeyFromSeed` from the API.